Orleans.Providers.MongoDB/Reminders/Store/MongoReminderCollection.cs

```diff
@@ -35,48 +37,71 @@ protected override string CollectionName()
                 {
-                    collection.Indexes.CreateOne(new CreateIndexModel<MongoReminderDocument>(byHashDefinition));
+                    throw;
                 }
+                
+                // ignore all exceptions that a pre-existing index has already been created. These indexes may
+                // have already been placed by Mongo Database administrators (e.g. "QueryOptimization_Finding1").
+                // in the future, if need to be culled, only try collect library generated indexes.
             }
 
-            var byNameDefinition =
-                Index
-                    .Ascending(x => x.IsDeleted)
-                    .Ascending(x => x.ServiceId)
-                    .Ascending(x => x.GrainId)
-                    .Ascending(x => x.ReminderName);
-            try
+            List<string> indexesToRemove =
+            [
+                // ByHash definitions.
+                "ByHash",
+#pragma warning disable CS0618 // Type or member is obsolete
+                $"{collection.GetFieldName(r => r.IsDeleted)}_1"
+                + $"_{collection.GetFieldName(r => r.ServiceId)}_1"
+                + $"_{collection.GetFieldName(r => r.GrainHash)}_1"
+#pragma warning restore CS0618 // Type or member is obsolete
+            ];
+
+            if (removeAllLegacyIndexes)
             {
-                collection.Indexes.CreateOne(
-                   new CreateIndexModel<MongoReminderDocument>(byNameDefinition,
-                        new CreateIndexOptions
-                        {
-                            Name = "ByName"
-                        }));
+                // these indexes are safe to keep and redundant.
+                // However, if a rolling deployment is made with silos with Orleans.Providers.MongoDB on version <=9.3.0,
+                // there is a high chance of collection scans impacting the full cluster. 
+                // See: https://github.com/OrleansContrib/Orleans.Providers.MongoDB/pull/154
+                indexesToRemove.AddRange([
+                    // ByName definition
+                    "ByName",
+#pragma warning disable CS0618 // Type or member is obsolete
+                    $"{collection.GetFieldName(r => r.IsDeleted)}_1"
+                    + $"_{collection.GetFieldName(r => r.ServiceId)}_1"
+                    + $"_{collection.GetFieldName(r => r.GrainId)}_1"
+                    + $"_{collection.GetFieldName(r => r.ReminderName)}_1",
+#pragma warning restore CS0618 // Type or member is obsolete
+                ]);
             }
-            catch (MongoCommandException ex)
+
+            foreach (var indexToRemove in indexesToRemove)
             {
-                if (ex.CodeName == "IndexOptionsConflict")
+                // best effort drop
+                try
+                {
+                    collection.Indexes.DropOne(indexToRemove);
+                }
+                catch (MongoCommandException)
```

Comment:
Please remove the exception type from the catch and update the comment to inline the reasoning here instead of the need to reference an external link:
```
catch
{
   // Ignore since failure to drop a legacy index should not affect Silo operations under any circumstances.
}
```